src/android/CameraPreviewFragment.java

```diff
@@ -222,27 +231,72 @@ public void deviceHasUltraWideCamera(HasUltraWideCameraCallback hasUltraWideCame
         hasUltraWideCameraCallback.onResult(defaultCamera == true && ultraWideCamera == true);
     }
 
-    public static Size calculateResolution(Context context, int targetSize) {
-        Size calculatedSize;
-        if (getScreenOrientation(context) == Configuration.ORIENTATION_PORTRAIT) {
-            calculatedSize = new Size((int) ((float) targetSize / ratio), targetSize);
-        } else {
-            calculatedSize = new Size(targetSize, (int) ((float) targetSize / ratio));
+     public static Size calculateResolution(Context context, int desiredWidthPx, double aspectRatio) {
+        int minWidthPx = 700;
+        // 1) get all supported JPEG output sizes
+        Size[] supportedSizes = getSupportedResolutions(context, CameraSelector.LENS_FACING_BACK);
+
+        // 2) if none available, fall back to a simple ratioWidth:ratioHeight rectangle
+        if (supportedSizes.length == 0) {
+            int fallbackHeight = Math.round(
+                    desiredWidthPx / (float) aspectRatio
+            );
+            return new Size(desiredWidthPx, fallbackHeight);
+        }
+
+        int ratioWidth  = 4;
```

Comment:
```suggestion
        double ratioWidth  = 4;
```

Can we test this for the height also